lib/marked.esm.js

```diff
@@ -1415,11 +1424,14 @@ function mangle(text) {
  */
 var Lexer_1 = class Lexer {
   constructor(options) {
+    this.setOptions(options);
+  }
+
+  setOptions(options) {
     this.tokens = [];
     this.tokens.links = Object.create(null);
     this.options = options || defaults$2;
-    this.options.tokenizer = this.options.tokenizer || new Tokenizer_1();
-    this.tokenizer = this.options.tokenizer;
+    this.tokenizer = this.options.tokenizer || Tokenizer_1.getStaticTokenizer();
```

Comment:
Just cleaning up these two lines. I didn't see a reason for them to be separated, but would appreciate a double-check.

Reply:
I feel like those lines nullify the need for the static tokenizer. With those two lines a new Tokenizer is only created once since it is set in options and reused every time marked is run after that.

Reply:
Ah that makes sense. So, it didn't necessarily do anything useful when Lexer was remade from scratch each time, but with a static Lexer, `this.options.tokenizer` also stores the Tokenizer between calls to `marked()`. 

Although, it still seems weird to me to use this intermediate step. What do you think about this, so we set the tokenizer in `this.tokenizer` and we don't have to reassign it from `this.options` each time?:

```
if(!this.tokenizer) {
    this.tokenizer = this.options.tokenizer || new Tokenizer();
}
```

Reply:
> So, it didn't necessarily do anything useful

Since `this.options` is the `defaults` object when no options are given then `defaults.tokenizer` is set here so the tokenizer is reused even on `new Lexer()`

Reply:
I don't think the static tokenizer is need because of this. It might still be helpful to save the lexer though.

Reply:
Removed Static Tokenizer.